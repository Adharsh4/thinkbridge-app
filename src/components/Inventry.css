.Inventry{
	padding: 2% 2% 1% 2%;
	width: 60%;
	margin: 3% auto;
	box-shadow: 0 0 5px 2px rgb(207, 193, 193);
	background-color: white;
}

.Inventry:hover{
	box-shadow: 0 0 5px 2px rgb(173, 158, 158);
	transition: background 0.2s, border 0.2s, border-radius 0.2s, box-shadow 0.2s;
	-webkit-transition: background 0.2s, border 0.2s, border-radius 0.2s, box-shadow 0.2s;
	-moz-transition: background 0.2s, border 0.2s, border-radius 0.2s, box-shadow 0.2s;
	-ms-transition: background 0.2s, border 0.2s, border-radius 0.2s, box-shadow 0.2s;
	-o-transition: background 0.2s, border 0.2s, border-radius 0.2s, box-shadow 0.2s;
}

@media only screen and (max-width:1100px) and (min-width:651px){
	.Inventry{
		width: 80%;
		
	}
}
@media only screen and (max-width:650px) and (min-width:100px){
	.Inventry{
		width: 90%;
		
	}
}

.loading{
	display: flex;
	flex-direction: column;
	align-items: center;
}



.inputField{
	width: 100%;
	border: none;
	border-bottom: 1px solid black;
}

.inputField:focus{
	border: none;
	outline: none;
	border-bottom: 2px solid rgb(131, 131, 224)
}

.row{
	margin-bottom: 4%;
}
@media only screen and (max-width:770px) and (min-width:451px){
	.row{
		margin-bottom: 8%;
	}
}

@media only screen and (max-width:450px) and (min-width:100px){
	.row{
		margin-bottom: 15%;
	}
}


.previewImage{
	width: 150px;
	height: 150px;
}

.add-product-button{
	display: flex;
	justify-content: center;
}

.image-preview{
	display: flex;
	justify-content: center;
	margin-bottom: 5%;
	margin-top: 2%;
}

.update-buttons{
	display: flex;
	justify-content: space-around;
}