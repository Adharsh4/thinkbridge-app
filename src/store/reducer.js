import * as actionTypes from './actionTypes';

const initialState = {
    products: [],
    isLoading: true,
    errorMessage: ""
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case actionTypes.START_FETCHING:
            return {
                ...state,
                isLoading: true,
                errorMessage: ""
            }
        case actionTypes.STOP_FETCHING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.SAVE_PRODUCTS:
            return {
                ...state,
                products: action.products,
                isLoading: false,
                selectedProduct: [],
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADINGTOTRUE:
            return {
                ...state,
                isLoading: true,
            }
        case actionTypes.SET_ERRORMESSAGE:
            return {
                ...state,
                errorMessage: action.message,
                isLoading: false
            }
        default:
            return state;
    }
}

export default reducer;