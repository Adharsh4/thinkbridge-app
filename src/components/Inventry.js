import React, { useState, useEffect } from "react";
// import { useStateValue } from "../utility/StateProvider";
import "./Inventry.css";
import {connect} from 'react-redux';
import { withRouter } from "react-router-dom";
import FileBase64 from 'react-file-base64';
import * as actions from "../store/actions"
import Loader from "react-loader-spinner"

function Inventry(props) {

    const [name, setName] = useState("")
    const [description, setDescription] = useState("")
    const [price, setPrice] = useState("")
    const [category, setCategory] = useState("mobiles")
    const [image, setImage] = useState("")
    const [email, setEmail] = useState(localStorage.getItem("email"))
    const [productId, setProductId]  = useState(localStorage.getItem("productId"));
    // const [isLoadingTemp, setIsLoadingTemp] = useState(false);

    useEffect(() => {
        if(productId && props.products.length === 0){
            props.getAllProducts(email)
        }else{
            props.changeIsLoading();
        }
    }, [])

    useEffect(() => {
        if(productId && props.products.length !== 0){
            let temp = props.products.filter((product) => product.id === productId)
            setName(temp[0].name);
            setDescription(temp[0].description);
            setPrice(temp[0].price);
            setCategory(temp[0].category);
            setImage(temp[0].image);
        }
    }, [props.products])

    const deleteProductHandler = (e) => {
        e.preventDefault();
        const isConfirm = window.confirm("Please comfirm")
        if(isConfirm){
            props.deleteProduct(productId, email);
            props.history.push("/showProducts")
        }
    }

    const addProductHandlerInInventry = (e, type) => {
        e.preventDefault();
        if(name.trim() === "" || description.trim() === "" || price.trim() === ""){
            const errorMessage = "Please fill all fields"
            props.setErrorMessage(errorMessage)
            alert(errorMessage);
            return;
        }
        if(image.trim() === ""){
            const errorMessage = "Please upload image in small size"
            props.setErrorMessage(errorMessage)
            alert(errorMessage);
            return;
        }
        if(isNaN(price)){
            const errorMessage = "Please enter valid price"
            props.setErrorMessage(errorMessage)
            alert(errorMessage);
            return;
        }
        const product = {
            name,
            description,
            price,
            category,
            image
        }
        if(type === "update"){
            props.updateProduct({
                product, 
                email: email,
                productId: productId
            })    
        }else{
            props.addProduct({
                product, 
                email: email
            })    
        }
        if(!props.isLoading){
            props.history.push("/showProducts")
        }
    }
    

    return (
        props.isLoading ? <div className="loading"><h3>Please wait...</h3><Loader type = "Oval" color="#00BFFF" height={30} width={30}/></div> :
        <div className="Inventry" data-test="component-Inventry">
            <div className="row">
                <div className="col-md-3">
                    <label htmlFor="name"><strong>Product Name</strong></label>
                </div>
                <div className="col-md-8">
                    <input type="text" className="inputField" value={name} onChange={(e) => setName(e.target.value)} placeholder="Product Name" id="name" name="name"/>
                </div>
            </div>
            <div className="row">
                <div className="col-md-3">
                    <label htmlFor="description"><strong>Product Description</strong></label>
                </div>
                <div className="col-md-8">
                    <input type="text" className="inputField" value={description} onChange={(e) => setDescription(e.target.value)} placeholder="Product Description" id="description" name="description"/>
                </div>
            </div>
            <div className="row">
                <div className="col-md-3">
                    <label htmlFor="price"><strong>Product Price</strong></label>
                </div>
                <div className="col-md-8">
                    <input type="text" className="inputField" value={price} onChange={(e) => setPrice(e.target.value)} placeholder="Product Price" id="price" name="price"/>
                </div>
            </div>
            <div className="row">
                <div className="col-md-3">
                    <label htmlFor="category"><strong>Product Category</strong></label>
                </div>
                <div className="col-md-8">
                    <select class="form-select" onChange={(e) => setCategory(e.target.value)}>
                        <option value="mobiles" defaultValue>Mobiles</option>
                        <option value="laptops">Laptops</option>
                        <option value="monitors">Monitors</option>
                        <option value="earphones">Earphones</option>
                    </select>
                    {/* <input type="text" className="inputField" value={category} onChange={(e) => setCategory(e.target.value)} placeholder="Product Category" id="category" name="category"/> */}
                </div>
            </div>
            {/* <pre>{JSON.stringify(item, null, '\t')}</pre> */}
            <div className="row">
                <div className="col-md-3">
                    <label htmlFor="image"><strong>Product Image</strong></label>
                </div>
                <div className="col-md-4">
                <FileBase64
                    type="file"
                    multiple={false}
                    onDone={({ base64 }) => setImage( base64 )}
                    />
                </div>
                {/* {image && <div className="col-md-4">
                    <img src={image} alt="sample Image" className="previewImage"></img>
                </div>} */}
            </div>
            {image && 
                <div className="image-preview">
                    <img src={image} alt="sample Image" className="previewImage"></img>
                </div>
            }
            {productId ? <div className="update-buttons">
                {/* <div className="col-md-3"> */}
                    <button className="btn btn-primary" onClick={deleteProductHandler}>Delete Product</button>
                {/* </div> */}
                {/* <div className="offset-md-1 col-md-3"> */}
                    <button className="btn btn-primary" onClick={(e) => addProductHandlerInInventry(e, "update")}>Update Product</button>
                {/* </div> */}
            </div> : 
                <div className="add-product-button">
                    <button className="btn btn-primary" onClick={(e) => addProductHandlerInInventry(e, "add")}>Add Product</button>
                </div>
            }
            
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        isLoading: state.isLoading,
        products: state.products,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addProduct: (productObject) => dispatch(actions.addProduct(productObject)),
        updateProduct: (productObject) => dispatch(actions.updateProduct(productObject)),
        deleteProduct : (product_id, email) => dispatch(actions.deleteProduct(product_id, email)),
        getAllProducts: (email) => dispatch(actions.getAllProducts(email)),
        changeIsLoading: () => dispatch(actions.changeIsLoading()),
        setErrorMessage: (message) => dispatch(actions.setErrorMessage(message))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Inventry));
