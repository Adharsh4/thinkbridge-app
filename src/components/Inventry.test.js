
import React from 'react';
import Inventry from "./Inventry"
import Enzyme, {shallow, mount} from 'enzyme'
import EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import {BrowserRouter} from 'react-router-dom'
import { createStore, applyMiddleware } from 'redux';
import reducer from "../store/reducer"
import thunk from 'redux-thunk';

Enzyme.configure({adapter: new EnzymeAdapter()})


describe('<Inventry /> unit test', () => {
    const mockStore = createStore(reducer, applyMiddleware(thunk));
    const getWrapper = () => mount(
      <Provider store={mockStore}>
          <BrowserRouter>
            <Inventry/>
        </BrowserRouter>
      </Provider>
    );
    const container = getWrapper();

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should have an name field', () => {
        expect(container.find('input[type="text"]').length).toEqual(4);
    });

    it('should have proper props for name field', () => {
        const nameField = container.find('input[type="text"]').at(0)
        nameField.simulate('change', {target: {value: 'mobile'}})
        expect(container.find('#name').props().value).toEqual('mobile')
    });

    it('should have proper props for description field', () => {
        const descriptionField = container.find('input[type="text"]').at(1)
        descriptionField.simulate('change', {target: {value: 'Good Mobile'}})
        expect(container.find('#description').props().value).toEqual('Good Mobile')
    });

    it('should have proper props for price field', () => {
        const priceField = container.find('input[type="text"]').at(2)
        priceField.simulate('change', {target: {value: '20000'}})
        expect(container.find('#price').props().value).toEqual('20000')
    });

    it('should have proper props for category field', () => {
        const categoryField = container.find('input[type="text"]').at(3)
        categoryField.simulate('change', {target: {value: 'mobile'}})
        expect(container.find('#category').props().value).toEqual('mobile')
    });

    it('should be able to click the Add product button', () => {
        const addProductButton = container.find('button')
        addProductButton.simulate('click')
    });
  
  });

