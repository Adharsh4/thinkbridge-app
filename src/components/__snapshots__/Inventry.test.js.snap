// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventry /> unit test should match the snapshot 1`] = `"<div class=\\"Inventry\\" data-test=\\"component-Inventry\\"><div class=\\"row\\"><div class=\\"col-md-3\\"><label for=\\"name\\"><strong>Product Name</strong></label></div><div class=\\"col-md-8\\"><input type=\\"text\\" class=\\"inputField\\" placeholder=\\"Product Name\\" id=\\"name\\" name=\\"name\\" value=\\"\\"></div></div><div class=\\"row\\"><div class=\\"col-md-3\\"><label for=\\"description\\"><strong>Product Description</strong></label></div><div class=\\"col-md-8\\"><input type=\\"text\\" class=\\"inputField\\" placeholder=\\"Product Description\\" id=\\"description\\" name=\\"description\\" value=\\"\\"></div></div><div class=\\"row\\"><div class=\\"col-md-3\\"><label for=\\"price\\"><strong>Product Price</strong></label></div><div class=\\"col-md-8\\"><input type=\\"text\\" class=\\"inputField\\" placeholder=\\"Product Price\\" id=\\"price\\" name=\\"price\\" value=\\"\\"></div></div><div class=\\"row\\"><div class=\\"col-md-3\\"><label for=\\"category\\"><strong>Product Category</strong></label></div><div class=\\"col-md-8\\"><input type=\\"text\\" class=\\"inputField\\" placeholder=\\"Product Category\\" id=\\"category\\" name=\\"category\\" value=\\"\\"></div></div><div class=\\"row\\"><div class=\\"col-md-3\\"><label for=\\"image\\"><strong>Product Image</strong></label></div><div class=\\"col-md-4\\"><input type=\\"file\\"></div></div><div class=\\"row\\"><div class=\\"offset-md-9\\"><button class=\\"btn btn-primary\\">Add Product</button></div></div></div>"`;
